clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1062,46 +1092,31 @@ invariant_checker_t::print_results()
 
 std::string
 invariant_checker_t::check_for_pc_discontinuity(
-    per_shard_t *shard, const memref_t &memref, const memref_t &prev_instr, addr_t cur_pc,
-    const std::unique_ptr<instr_autoclean_t> &cur_instr_decoded, bool expect_encoding,
+    per_shard_t *shard, const per_shard_t::instr_info_t &prev_instr_info,
+    const per_shard_t::instr_info_t &cur_instr_info, bool expect_encoding,
     bool at_kernel_event)
 {
+    const memref_t prev_instr = prev_instr_info.memref;
     std::string error_msg = "";
     bool have_branch_target = false;
     addr_t branch_target = 0;
-    const addr_t prev_instr_trace_pc = prev_instr.instr.addr;
-    if (prev_instr_trace_pc == 0 /*first*/)
+    addr_t prev_instr_trace_pc = prev_instr.instr.addr;
+    const addr_t cur_pc = at_kernel_event ? cur_instr_info.memref.marker.marker_value
```

Comment:
This is an assumption that should probably be documented (previously this cur_pc was passed in which seemed a little cleaner) -- that "cur_instr_info" really is a marker if "at_kernel_event".